scripts/word_embeddings/train_word2vec.py

```diff
@@ -0,0 +1,370 @@
+    group.add_argument('--epochs', type=int, default=5, help='Epoch limit')
+    group.add_argument('--gpu', type=int, nargs='+',
+                       help=('Number (index) of GPU to run on, e.g. 0. '
+                             'If not specified, uses CPU.'))
+    group.add_argument('--no-hybridize', action='store_true',
+                       help='Disable hybridization of gluon HybridBlocks.')
+    group.add_argument(
+        '--no-static-alloc', action='store_true',
+        help='Disable static memory allocation for HybridBlocks.')
+    group.add_argument('--no-sparse-grad', action='store_true',
+                       help='Disable sparse gradient support.')
+
+    # Model
+    group = parser.add_argument_group('Model arguments')
+    group.add_argument('--emsize', type=int, default=300,
+                       help='Size of embedding vectors.')
+    group.add_argument('--model', type=str, default='skipgram',
+                       help='SkipGram or CBOW.')
+    group.add_argument('--window', type=int, default=5,
+                       help='Context window size.')
+    group.add_argument('--negative', type=int, default=5,
+                       help='Number of negative samples.')
+
+    # Optimization options
+    group = parser.add_argument_group('Optimization arguments')
+    group.add_argument('--optimizer', type=str, default='adagrad')
+    group.add_argument('--lr', type=float, default=0.1)
+    group.add_argument('--elementwise-clip-gradient', type=float, default=-1,
+                       help='Clip embedding matrix gradients elementwise. '
+                       'Disable by setting to a value <= 0.')
+    group.add_argument(
+        '--groupwise-clip-gradient', type=float, default=-1,
+        help='Clip embedding matrix gradients '
+        'such that the norm of the gradient for one embedding vector '
+        'does not surpass --groupwise-clip-gradient.'
+        'Disable by setting to a value <= 0.')
+
+    # Logging
+    group = parser.add_argument_group('Logging arguments')
+    group.add_argument('--logdir', type=str, default='logs',
+                       help='Directory to store logs.')
+    group.add_argument('--eval-interval', type=int, default=10000)
+    group.add_argument('--eval-analogy', action='store_true')
+
+    # Evaluation options
+    evaluation.add_parameters(parser)
+
+    args = parser.parse_args()
+    evaluation.validate_args(args)
+    return args
+
+
+def get_train_data():
+    """Helper function to get training data."""
+    with print_time('load training dataset'):
+        dataset = nlp.data.Text8(segment='train')
+
+    with print_time('count tokens'):
+        counter = nlp.data.count_tokens(itertools.chain.from_iterable(dataset))
+
+    vocab = nlp.Vocab(counter, unknown_token=None, padding_token=None,
+                      bos_token=None, eos_token=None, min_freq=5)
+
+    negatives_weights = mx.nd.array(
+        [counter[w] for w in vocab.idx_to_token])**0.75
```

Comment:
Should 0.75 be part of UnigramCandidateSampler's argument? We can add an option like "distortion"

Reply:
I think it's ok to keep separate, given that the distortion is simple to apply and there may be other distortions in other problems.